addons/binding/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/handler/RobonectHandler.java

```diff
@@ -0,0 +1,460 @@
+                ErrorEntry lastErrorEntry = errorList.getErrors().get(0);
+                updateLastErrorChannels(lastErrorEntry);
+            }
+        }else {
+            logger.debug("Could not retrieve mower error list. Robonect error response message: {}",
+                                errorList.getErrorMessage());
+        }
+    }
+
+    private void updateLastErrorChannels(ErrorEntry error) {
+        if (error.getErrorMessage() != null) {
+            updateState(CHANNEL_LAST_ERROR_MESSAGE, new StringType(error.getErrorMessage()));
+        }
+        if (error.getErrorCode() != null) {
+            updateState(CHANNEL_LAST_ERROR_CODE, new DecimalType(error.getErrorCode().intValue()));
+        }
+        if (error.getDate() != null) {
+            State dateTime = convertDateTimeType(error.getDate(), error.getTime());
+            updateState(CHANNEL_LAST_ERROR_DATE, dateTime);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        try {
+            RobonectConfig robonectConfig = getConfigAs(RobonectConfig.class);
+            RobonectEndpoint endpoint = new RobonectEndpoint(robonectConfig.getHost(), robonectConfig.getUser(),
+                    robonectConfig.getPassword());
+            httpClient = new HttpClient();
+            httpClient.start();
+            robonectClient = new RobonectClient(httpClient, endpoint);
+            refreshVersionInfo();
+            Runnable runnable = new Runnable() {
+                
+                private long offlineSince = -1;
+                private long offlineTriggerDealay = robonectConfig.getOfflineTimeout() * 60 * 1000;
+                private boolean offlineTimeoutTriggered = false;
+                
+                @Override
+                public void run() {
+                    try {
+                        refreshMowerInfo();
+                        updateStatus(ThingStatus.ONLINE);
+                        offlineSince = -1;
+                        offlineTimeoutTriggered = false;
+                    } catch (RobonectCommunicationException rce) {
+                        if(offlineSince < 0){
+                            offlineSince = System.currentTimeMillis();
+                        }
+                        if(!offlineTimeoutTriggered && System.currentTimeMillis() - offlineSince > offlineTriggerDealay){
+                            // trigger offline
+                            updateState(CHANNEL_MOWER_STATUS_OFFLINE_TRIGGER, new StringType("OFFLINE_TIMEOUT") );
+                            offlineTimeoutTriggered = true;
+                        }
+                        logger.debug("Failed to communicate with the mower. Taking it offline.", rce);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, rce.getMessage());
+                    } catch(com.google.gson.JsonSyntaxException jse){
```

Comment:
Can you import the `com.google.gson.JsonSyntaxException`

Reply:
of course :-)